src/wavm/openmp.cpp

```diff
@@ -811,24 +829,23 @@ int reduceFinished()
  */
 void finaliseReduce(bool barrier)
 {
-    auto logger = faabric::util::getLogger();
-    auto ctx = threads::getOpenMPContext();
-
-    // Master must make sure all other threads are done
-    if (ctx.threadNumber == 0) {
-        ctx.level->masterWait(ctx.threadNumber);
-
-        logger->debug("Master thread finished reduce");
-    }
+    threads::OpenMPContext& ctx = threads::getOpenMPContext();
 
     // Everyone waits if there's a barrier
     if (barrier) {
+        PROF_START(FinaliseReduceBarrier)
         ctx.level->barrier.wait();
+        PROF_END(FinaliseReduceBarrier)
```

Comment:
Also a good point to watch for performance issues.

Reply:
:+1: